lib/Conversion/TorchToLinalg/DataMovement.cpp

```diff
@@ -301,6 +301,42 @@ class ConvertAtenViewOp : public OpConversionPattern<AtenViewOp> {
           op, "desired size list length mismatches with the result type rank");
     }
 
+    // Case where all sizes (input and output) are statically known
```

Comment:
Why are we doing this? I thought we could flatten to 1d and then expand in the worst case?

Reply:
I don't think I understand the question - that's what the code here is doing, isn't it? Or do you mean this should only happen when the other cases aren't applicable?

Reply:
There should be no static shape limitation.

Reply:
I didn't think that collapsing from `[?,?] -> [?]` was supported but I can give it a try.

Reply:
Okay, I've refactored it a bit so it doesn't need a static input shape. Expanding `[?]` to either all static or all dynamic output shapes isn't supported, so I don't think we can do the collapse -> expand thing in an entirely dynamic context.

Reply:
@MaheshRavishankar FYI for expanding with dynamic dimensions

Reply:
@silvasean Should this get merged then, given that the expansion isn't possible in a dynamic context?

Reply:
Thanks Sean... Still dont have an owner for `tensor.expand_shape` with dynamic expanded dimensions. @gpetters94 just to clarify, `[?, ?] -> [?]` is supported with `tensor.collapse_shape` today.

Reply:
@gpetters94 ok let's merge it, but please file a bug on llvm/llvm-project for this and link it from a TODO in the code.

Reply:
Hold on I still have a problem here. This should not be doing a full flatten by default for the static dim case. We only want to do it if the other analysis fails. This seems to regress other static support that we have, assuming I correctly read how the code here is working. The correct approach should be in #1353 which constructs the reassociation indices for expand/collapse considering both dynamic and static shapes. The previous version of #1353 was reverted for issues with dynamic shape support as well as unit tests but that PR has the exact same problem with tensor.expand_shape.

The canonicalization here still makes sense however. #1353 needs to check with this PR to make sure all of the cases here are handled with the canonicalization in place.

Reply:
@qedawkins @silvasean Do you think a refactor so that the logic in this PR runs only if all the other cases fail would work?